### ISSUE TITLE
Allow to update sets of contents based on match criteria

### ISSUE THREAD

**Issue by gggeek** [kaliop-uk/ezmigrationbundle#18]
This makes updates easier to write than merely adding a list of IDs as an array

Eg: update attribute X to value Y of all locations which are child of node Z and have attribute W set to value K

**Comment by gggeek** [comment]
This is blocked by issue #17

**Comment by gggeek** [comment]
How to go at it:

- for 'update' and 'delete' actions, for both content and location, add a 1st-level 'match' keyword
- under that, we can use keywords: object_id (single or array), object_remote_id (single or array), etc...
- the existing 1st level keywords we need to support, but we should deprecate them

**Comment by gggeek** [comment]
Next steps:

- remove the location_resolver stuff, replacing it with matchers across all handlers

**Comment by gggeek** [comment]
WIP...

**Comment by gggeek** [comment]
WIP...

**Comment by gggeek** [comment]
The base is there; most likely widespread usage will be pushed to rev. 2.1 or 3...

**Comment by gggeek** [comment]
The 'matcher' concept should be extended as well to users (match by email, login), user-groups, content-types, roles

**Comment by gggeek** [comment]
Another usecase: delete all contents of type X before deleteing the contenttype itself

**Comment by gggeek** [comment]
updating and deleting of contents based on match conditions: done

**Comment by gggeek** [comment]
Closing. Opening a separate issue for widespread usage of matchers
